test/gurka/test_route_incidents.cc

```diff
@@ -0,0 +1,685 @@
+    return incident_tile_;
+  }
+  void sort() {
+    std::sort(incident_tile_->begin(), incident_tile_->end(),
+              [](const GraphReader::Incident& a, const GraphReader::Incident& b) {
+                if (a.edge_index == b.edge_index) {
+                  if (a.start_offset == b.start_offset) {
+                    if (a.end_offset == b.end_offset) {
+                      return a.id < b.id;
+                    }
+                    return a.end_offset < b.end_offset;
+                  }
+                  return a.start_offset < b.start_offset;
+                }
+                return a.edge_index < b.edge_index;
+              });
+  }
+  std::shared_ptr<baldr::GraphReader::incident_tile_t> incident_tile_;
+};
+
+// used to make a graphreader and mark some edges as having incidents
+std::shared_ptr<test_reader> setup_test(const gurka::map& map,
+                                        const std::vector<std::string>& names,
+                                        std::vector<baldr::GraphId>& edge_ids) {
+  // make our reader that we can manipulate
+  auto reader = std::make_shared<test_reader>(map.config.get_child("mjolnir"));
+
+  // modify traffic speed info to say that this edge has an incident
+  for (const auto& name : names) {
+    auto begin_node = name.substr(0, 1);
+    auto end_node = name.substr(1);
+    auto edge_id = std::get<0>(gurka::findEdgeByNodes(*reader, map.nodes, begin_node, end_node));
+    auto has_incident_cb = [edge_id](baldr::GraphReader& reader, baldr::TrafficTile& tile,
+                                     int edge_index, valhalla::baldr::TrafficSpeed* current) -> void {
+      if (edge_id.Tile_Base() == tile.header->tile_id && edge_id.id() == edge_index)
+        current->has_incidents = true;
+    };
+    valhalla_tests::utils::customize_live_traffic_data(map.config, has_incident_cb);
+    edge_ids.push_back(edge_id);
+  }
+
+  // prepare for incidents
+  reader->incident_tile_.reset(new decltype(reader->incident_tile_)::element_type);
+  return reader;
+}
+
+// via some macro magic we are inside the scope of the incidents class above
+TEST_F(incidents, simple_cut) {
+  // mark the edges with incidents
+  std::vector<baldr::GraphId> edge_ids;
+  auto reader = setup_test(map, {"CB"}, edge_ids);
+  std::shared_ptr<baldr::GraphReader> graphreader(reader.get(), [](baldr::GraphReader*) {});
+
+  // modify the incident tile to have incidents on this edge
+  reader->incident_tile_->emplace_back(
+      baldr::GraphReader::Incident{edge_ids[0].id(), .25, .75, 1234});
+
+  // do the route
+  auto result = gurka::route(map, {"C", "B"}, "auto",
+                             {{"/filters/action", "include"}, {"/filters/attributes/0", "incidents"}},
+                             graphreader);
+
+  // check its right
+  check_incident_locations(result, {
+                                       {1234, 0, 0, .25, 0, .75},
+                                   });
+}
```

Comment:
this is the basic layout of each test. turn on incidents for your edgeids, make the incidents in the incident tile, get a route with incidents turned on. then reason about what the incident associated to the leg will look like and check that it looks that way